examples/DynamicProperties-sample.xml

```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
+  <edmx:Reference Uri="https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Aggregation.V1.xml">
+    <edmx:Include Namespace="Org.OData.Aggregation.V1" Alias="Aggregation" />
+  </edmx:Reference>
+  <edmx:Reference Uri="https://sap.github.io/odata-vocabularies/vocabularies/UI.xml">
+    <edmx:Include Namespace="com.sap.vocabularies.UI.v1" Alias="UI" />
+  </edmx:Reference>
+  <edmx:Reference Uri="https://sap.github.io/odata-vocabularies/vocabularies/Analytics.xml">
+    <edmx:Include Namespace="com.sap.vocabularies.Analytics.v1" Alias="Analytics" />
+  </edmx:Reference>
+  <edmx:DataServices>
+    <Schema Namespace="DynamicProperties.examples" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+
+      <EntityType Name="Sales">
+        <Key>
+          <PropertyRef Name="Order"/>
+        </Key>
+        <Property Name="Order" Type="Edm.String" Nullable="false"/>
+        <Property Name="Country" Type="Edm.String" Nullable="false"/>
+        <Property Name="NumberOfItems" Type="Edm.Int16" Nullable="false"/>
+        <Property Name="Amount" Type="Edm.Decimal"/>
+        <Annotation Term="Analytics.AggregatedProperties" Qualifier="sum">
+          <Collection> <!-- collection always has one entry that can be referenced with index segment /0 -->
+            <Record>
+              <PropertyValue Property="Name" String="Total"/>
+              <PropertyValue Property="AggregationMethod" String="sum"/>
+              <PropertyValue Property="AggregatableProperty" PropertyPath="Amount"/>
+            </Record>
+          </Collection>
+        </Annotation>
+        <Annotation Term="Analytics.AggregatedProperties" Qualifier="max">
+          <Collection>
+            <Record>
+              <PropertyValue Property="Name" String="Maximum"/>
+              <PropertyValue Property="AggregationMethod" String="max"/>
+              <PropertyValue Property="AggregatableProperty" PropertyPath="Amount"/>
+            </Record>
+          </Collection>
+        </Annotation>
+        <Annotation Term="UI.Chart">
+          <Record>
+            <PropertyValue Property="ChartType" EnumMember="UI.ChartType/Bubble"/>
+            <PropertyValue Property="Dimensions">
+              <Collection>
+                <PropertyPath>Country</PropertyPath>
+              </Collection>
+            </PropertyValue>
+            <PropertyValue Property="Measures">
+              <Collection>
+                <PropertyPath>NumberOfItems</PropertyPath>
+              </Collection>
+            </PropertyValue>
+            <PropertyValue Property="DynamicMeasures">
+              <Collection>
+                <AnnotationPath>@Analytics.AggregatedProperties#sum/0</AnnotationPath>
```

Comment:
The [OData spec](https://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#sec_AnnotationPath) says:
>A non-null path MUST resolve to an annotation.

Does that mean that it cannot resolve to a property or a collection item _within_ an annotation?

Reply:
Correct, to reference a property within an annotation use `Edm.PropertyPath` or `Edm.NavigationPropertyPath`

Reply:
And to cover everything, use an `Edm.ModelElementPath`. Can we even relax the existing type in `UI.Chart/Measures` from `Edm.PropertyPath` to `Edm.ModelElementPath` without breaking compatibility?

Reply:
> use an `Edm.ModelElementPath`

This would require a new vocabulary (version) using OData-CSDL-Version 4.01, which we so far avoided.

Reply:
Good to know this boundary condition.